server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -393,6 +399,9 @@ public Decision canRemain(IndexMetadata indexMetadata, ShardRouting shardRouting
         if (indexMetadata.ignoreDiskWatermarks()) {
             return YES_DISK_WATERMARKS_IGNORED;
         }
+        if (useAverageDiskUsage(node, usages) == false) {
+            return YES_AVERAGE_DISK_USAGE_UNAVAILABLE;
+        }
```

Comment:
This is added before every call `getDiskUsage` that might compute the average.
Is there a value of making `getDiskUsage` return null if overflow happens and have it as a signal to return above decision? This way the addition of total and free could be performed only once (in `getDiskUsage`) vs twice (in `getDiskUsage` and the check above).